wordpress-mobile/WordPress-FluxC-Android
# [Detekt Baseline Warnings] Plugins Woo Commerce Module - Resolve/Suppress Style Warnings

Parent #2479
Partially Closes: #2486

This PR resolves/suppresses all `style` related warnings for the `plugins:woocommerce` module:

1. `3` x DataClassShouldBeImmutable (Resolve: 8a526d4ab273cdaf9dcc5cc8e3b1383e8375de5a + 44788816a8ce9d79b2dc4b7768662dd8bae8141f)
2. `6` x ForbiddenComment (Suppress: b7466eee836f7cb2ec1d4cdce174a8c490614cd8)
3. `15` x MagicNumber (Resolve: f55ad6e65a38b9eea888b3b91ee9a02a86e5ba01 + Suppress: c33859f36ade917667213484c86b8e255d777b64 + Question [❓]: [see comment](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1103#issuecomment-1211868401) @anitaa1990 @AmandaRiu)
4. `9` x MaxLineLength (Resolve: 5d6b66222333d6410317cfc0c8b04c32c3db7044 + Suppress: 7097dc2af36808c427132075a85486990d3248a4)
5. `13` x ReturnCount (Resolve: c700ef6f7945ff6b739346e259b490412fc03d33 + Suppress: 2e9394667112549e535a3cb165d510f9fab5bf09)
6. `1` x SpacingBetweenPackageAndImports (Resolve: 6dcca837a736ada45008ad8f9dc22b931a8a126d)
7. `2` x UnnecessaryAbstractClass (Resolve: 377e3ca9ff7d9208a81b05d88bbde0fe13ddf9ca)
8. `1` x UnusedPrivateMember (Resolve: 6871985de7f76ba4f137fc6a3dd55c23c5c4135a)
9. `2` x UseCheckOrError (Resolve: 1594872df7959dab2af33114b5ff253ce9934e9c + Suppress: 5236ee56acb95d259a0027c4eef47a074de04e02)

PS.1: I am randomly adding an engineer from the WCAndroid team as the main reviewers, that is, in addition to the @wordpress-mobile/owl-team team itself, since I just want someone from that team to sign-off on that change as well (or additionally to the 🦉 team).

PS.2: I recommend reviewing this PR commit-by-commit.

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough (especially the `detekt` check).
- However, if you really want to be thorough, you could smoke test the `example` and/or the `WCAndroid` apps to verify that everything works as expected.

PS: Please ignore the failing `connected-tests` check as they are unrelated to this PR. This check has been failing consistency for some time now due to tests being flaky or needing to be updated/fixed.
